reproducible/Dockerfile.nix

```diff
@@ -0,0 +1,185 @@
+            ''; \n\
+\n\
+        fixupPhase = '' \n\
+                echo \"fixup phase skipped \" \n\
+            ''; \n\
+        \n\
+        shellHook = '' \n\
+\n\
+                cd \$out\n\
+                chmod -Rf a+w .\n\
+\n\
+                echo \"=== ctest -E samples\|oegdb-test\|report\" \n\
+                if \$DO_CHECK\n\
+                then \n\
+                    find ./tests -type d -exec chmod a+w {} \;\n\
+\n\
+                    LD_LIBRARY_PATH=/home/azureuser/.nix_libs \$OE_SIM ctest -E samples\|oegdb-test\|report \n\
+                    CTEST_RESULT=\$?\n\
+                    if [ \$CTEST_RESULT -ne 0]\n\
+                    then\n\
+                        echo \"ERROR: Ctests failed\"\n\
+                        exit \$CTEST_RESULT\n\
+                    fi\n\
+                fi\n\
+\n\
+                if [ \$(uname -m) == \"aarch64\" ]\n\
+                then \n\
+                    LD_INTERPRETER=\"/lib/aarch64-linux-gnu/ld-linux-aarch64.so.1\"\n\
+                elif [ \$(uname -m) == \"x86_64\" ]\n\
+                then\n\
+                    LD_INTERPRETER=\"/lib64/ld-linux-x86-64.so.2\"\n\
+                else\n\
+                    LD_INTERPRETER=\"UNSUPPORTED ARCHITECTURE\"\n\
+                fi\n\
+                echo \"=== FIXUP \$LD_INTERPRETER\"\n\
+                find \$out -type f -executable -exec patchelf --set-interpreter \$LD_INTERPRETER {} \;\n\
+                find \$out -type f -executable -exec patchelf --remove-rpath {} \;\n\
+\n\
+		if [ "$DEB_PACKAGE" != 'false' ] \n\
+                then\n\
+                    /home/azureuser/build_deb_pkg.sh\n\
+                else \n\
+                    echo "skipping package build" \n\
+                fi\n\
+                if \$INTERACTIVE_SHELL\n\
+                then\n\
+                    echo \"=== Complete\"\n\
+                    exit 0\n\
+                fi \n\
+            '';  \n\
+}  \n\
+" > /home/$BUILD_USER/shell.nix
+
+
+
+RUN echo "User is $USER "
+#install the required software
+#RUN touch .bash_profile \
+#
+# We add the nix install and packages into the container rather than waiting for run time.
+# The packages are then located in the nix store until the next push of the container
+#RUN curl https://nixos.org/releases/nix/nix-2.3.7/install | /bin/bash
+ADD ./install-nix.sh /home/$BUILD_USER
+RUN  /bin/bash /home/$BUILD_USER/install-nix.sh
+ENV NIX_PATH=/home/$BUILD_USER
+ADD ./prep-nix-build.sh /home/$BUILD_USER
+RUN /bin/bash ./prep-nix-build.sh /home/$BUILD_USER/nixpkgs
+
+ADD ./sort_deb_sum.sh /home/$BUILD_USER
+ADD ./nix-build.sh /home/$BUILD_USER
+ADD ./nix-shell.sh /home/$BUILD_USER
+ADD ./build_deb_pkg.sh /home/$BUILD_USER
+RUN mkdir -p /home/$BUILD_USER/.nix_libs
+ADD ./nix-libs/libsgx_enclave_common.so /home/$BUILD_USER/.nix_libs
+ADD ./nix-libs/libsgx_enclave_common.so.1 /home/$BUILD_USER/.nix_libs
+ADD ./nix-libs/libsgx_launch.so.1  /home/$BUILD_USER/.nix_libs
```

Comment:
As [this feedback](https://github.com/openenclave/openenclave/pull/3744/files#r527924046) shows, libsgx_enclave_common is not required for OE SDK build. libsgx_launch, libprotobuf probably are not needed either. libstdc++ dependency is also unexpected.

Reply:
They are required to run tests within the container.

They are not required for the build proper, but running ctests is a requirement if we are to use this in a pipeline.
nix restricts access to os libraries, so it may seem strange, but a binary running even in nix shell cannot see the native os libraries.

Reply:
the .nix_lib contents are removed from the build container altogether and only added in the test container.